Rewrote Getting Help FAQ.

adamchainz opened the pull request:
Make it a more useful place to point beginners with questions.

adamchainz commented:
I've had a few random tweets recently with questions: https://twitter.com/code_In_ME/status/1241422676553601025 . Also pointing to the forum to increase its use and utility - google groups are a bit passé and worse UX.

adamchainz commented:
@andrewgodwin please also edit the [pinned Using Django post](https://forum.djangoproject.com/t/welcome-to-using-django/14) to point to this FAQ so users can read our advice on asking good questions?

andrewgodwin commented:
Done, though it points to the stable release right now - I can change it to point to dev if you're not going to backport this?

sks444 commented:
Hi @adamchainz, can we also add a link to the forum in the getting help section at https://docs.djangoproject.com/en/3.0/#getting-help ?

adamchainz commented:
@andrewgodwin Although we rarely backport docs changes, I think should backport this one, since we can't control which version users are on when they click "get help"

@sks444 Thanks for the prompt. I actually rewrote that section to point directly to this FAQ page, to reduce the redundant redundancy. It also linked directly to creating a ticket, rather than going through the "how to" of creating tickets as this FAQ page does. Hopefully that reduces frivolous tickets.

I also took a little extra pass at the wording. I renamed these to "channels" of support, added a mention of the Code of Conduct, and reduced some mailing-list centric language.

adamchainz commented:
Interested to see both fellows' input on this.

smithdc1 commented on docs/faq/help.txt:
```diff
@@ -5,15 +5,34 @@ FAQ: Getting Help
+.. _`"Using Django"`: https://forum.djangoproject.com/c/users
 .. _`#django IRC channel`: irc://irc.freenode.net/django
 
+In all these channels, please be friendly and patient, and otherwise abide by
```
I'm not so sure about ',and otherwise'. My view is it would be ok without the word 'otherwise'.
I think the oxford comma helps readability here.

adamchainz replied:
"Be friendly and patient" is the first point in the Code of Conduct, so it's a separate clause to "and otherwise..". I've rewritten it to split a slightly longer summary of the code afterwards, for those readers who won't click the link.

smithdc1 commented on docs/faq/help.txt:
```diff
@@ -30,23 +49,25 @@ that spammers get caught, but it also means that your first question to the
+You might notice we have a second mailing list, called |django-developers|.
+Please don't email support questions to this mailing list. This list is for
+discussion of the development of Django itself. Asking a tech support question
+there is considered impolite, and you will likely simply be directed to ask on
```
I think the use of 'impolite' is a little strong, and I think the ordering could be switched around a bit. Let's tell people what it's for and then say why tech support questions shouldn't be asked here. 

Here is my attempt. 

```
You might notice we have a second mailing list, called |django-developers|.
This list is for discussion of the development of Django itself. Please don't 
email support questions to this mailing list as you are likely to be 
directed to ask on |django-users|.
```

adamchainz replied:
Yes good point. I agree we should explain first.

I'm not sure "impolite" is that strong. A strong word would be "rude". Removing any such description feels too weak to me as it implies the only negative thing is that you'll be redirected.

Even with the current wording we still get maybe 1-5 posts a week on there asking support questions that I turn down with an auto-fill message.

carltongibson replied:
We could go all _Captain Hook_ and have _is considered bad form..._ (🙂 Not necessarily 100% serious... — but maybe.)

I keep tripping on the _"likely simply"_ — can we find a better phrasing for that?

adamchainz replied:
I simply dropped "simply"

carltongibson commented on docs/index.txt:
```diff
@@ -38,17 +38,8 @@ Having trouble? We'd like to help!
-.. _#django IRC channel: irc://irc.freenode.net/django
-.. _ticket tracker: https://code.djangoproject.com/
+* Not found anything? See :doc:`faq/help` for information on asking questions
+  to the community.
```
I'd suggest we labour the point a little here...

```suggestion
* Not found anything? See :doc:`faq/help` for information on getting support 
  and asking questions to the community.

```

... because I think we need to keep the link to Trac on the homepage. 

We could reword that last to emphasise "You've found a bug?", but TBH we don't get a lot of support requests on the issue tracker. (Some, yes, but...)

felixxm commented on docs/faq/help.txt:
```diff
@@ -5,15 +5,35 @@ FAQ: Getting Help
-try the |django-users| mailing list. Feel free to ask any question related
-to installing, using, or debugging Django.
+First, please check if your question is answered on the
+:doc:`FAQ index <faq/index>`. Please also search for answers using your
```
I would chop `index`, ``:doc:`FAQ <faq/index>` ``.

felixxm replied:
Too many "please" IMO, maybe: _"Also, search ..."_.

carltongibson replied:
I'll adjust these and merge.
